packages/core/src/utils.ts

```diff
@@ -619,36 +619,39 @@ export const readBuildInfo = (buildInfoPath: string): BuildInfo => {
     fs.readFileSync(buildInfoPath, 'utf8')
   )
 
-  validateBuildInfo(buildInfo)
```

Comment:
This function is only called from `getPreviousStorageLayoutOZFormat` which is called after all the build info files are checked anyway, so it's really not necessary to do this check and keeping it would add unnecessary complexity. We could honestly probably refactor `getPreviousStorageLayoutOZFormat` to remove the need for this function at all since it's the only place where we still read the build info files separately from `makeGetConfigArtifacts`.

Reply:
Sounds good, we can leave it as is. I don't think it makes sense to refactor the `readBuildInfo(previousBuildInfo)` call though. The`previousBuildInfo` is the build info file that was used in a previous deployment, not the current one, so I don't think it makes sense to add this to `getConfigArtifacts`. I still think it's fine not to include `validateBuildInfo` inside `readBuildInfo` though because it's probably safe to assume that the build info file contains the storage layouts.